src/shared/switch-root.c

```diff
@@ -37,7 +37,7 @@ int switch_root(const char *new_root,
                 { "/dev",                                 MS_BIND|MS_REC }, /* Recursive, because we want to save the original /dev/shm + /dev/pts and similar */
                 { "/sys",                                 MS_BIND|MS_REC }, /* Similar, we want to retain various API VFS, or the cgroupv1 /sys/fs/cgroup/ tree */
                 { "/proc",                                MS_BIND|MS_REC }, /* Similar */
-                { "/run",                                 MS_BIND        }, /* Stuff mounted below this we don't save, as it might have lost its relevance, i.e. credentials, removable media and such, we rather want that the new boot mounts this fresh */
+                { "/run",                                 MS_BIND|MS_REC }, /* Stuff mounted below this we don't save, as it might have lost its relevance, i.e. credentials, removable media and such, we rather want that the new boot mounts this fresh */
                 { SYSTEM_CREDENTIALS_DIRECTORY,           MS_BIND        }, /* Credentials passed into the system should survive */
                 { ENCRYPTED_SYSTEM_CREDENTIALS_DIRECTORY, MS_BIND        }, /* Similar */
```

Comment:
The comment is outdated now.

And it seems iffy to exclude SYSTEM_CREDENTIALS_DIRECTORY and  ENCRYPTED_SYSTEM_CREDENTIALS_DIRECTORY. Normally, we don't expect any mounts underneath those. But if there *are* any, then they should be propagated. Thus, I think this `transfer_table` should be made into a strv, and the same `MS_BIND|MS_REC` flags should be used everywhere.

While at it, please fix the adjective/adverb confusion for "Similar" and "Recursive".